logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -416,6 +423,7 @@ private void doCreateSnapshot() {
     if (currentEvent != null) {
       final long commitPosition = streamProcessorContext.getLogStream().getCommitPosition();
 
+      // TODO: delegate to snapshot controller?
```

Comment:
I haven't yet. I'm leaning towards the `StreamProcessorController` rather indicating to the snapshot controller that "you could take a snapshot now" and let the snapshot controller decide whether or not to do so...but that means introducing state into the snapshot controller. So I'm still on the fence. Any opinion?

In either case, it isn't required for this issue and I'd like to get RocksDB in, so it could be delayed to a next step. I can remove the TODO for now regardless.

Reply:
I think it would be good to move this into the snapshot controller.